### ISSUE TITLE
Error when running on .html files

### ISSUE THREAD

**Issue by aress31** [typescript-eslint/typescript-eslint#2529]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:lit/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "eslint-config-prettier"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "extraFileExtensions": [".html"],
    "tsconfigRootDir": ".",
    "project": ["./tsconfig.json"]
  },
  "plugins": [
    "@typescript-eslint",
    "import",
    "html",
    "sort-keys-fix"
  ],
  "root": true,
  "rules": {
    "import/no-unresolved": "off",
    "sort-keys-fix/sort-keys-fix": "error"
  }
}
```

```TS
// your repro code case
  "scripts": {
    "lint:eslint": "eslint \"**/*.{html,ts}\" --ignore-path .gitignore",
    --- SNIP ---
  }
```

```JSON
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
    "importsNotUsedAsValues": "error",
    "inlineSources": true,
    "lib": ["es2017", "dom"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "outDir": "out-tsc",
    "plugins": [
      {
        "name": "ts-lit-plugin",
        "strict": true
      }
    ],
    "removeComments": true,
    "resolveJsonModule": true,
    "rootDir": "./",
    "sourceMap": true,
    "strict": true,
    "target": "es2018",
    "typeRoots": ["custom_typings", "node_modules/@types"]
  },
  "exclude": [
    "node_modules",
    "build",
    "custom_typings/**/*.ts"
  ],
  "include": ["**/*.ts"]
}
```

**Expected Result**

When I run the command `yarn lint:eslint`, I am getting the following error message about the `index.html` file as it can be seen in the following output:
```
yarn lint:eslint --fix
yarn run v1.22.4
$ eslint "**/*.{html,ts}" --ignore-path .gitignore --fix

C:\Users\ateyar\Documents\GitHub\aegiscyber-wc\index.html
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: index.html.
The file must be included in at least one of the projects provided

C:\Users\ateyar\Documents\GitHub\aegiscyber-wc\src\app.ts
  138:5  error  Promises must be handled appropriately or explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

✖ 2 problems (2 errors, 0 warnings)

error Command failed with exit code 1.
```

I want to run `eslint` on `.html` files as well because some of them can embed `JavaScript` code. I am therefore wondering how I could configure my project to remove this error?
**Actual Result**

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^4.0.1` |
| `TypeScript`                | `^4.0.2` |
| `ESLint`                    | `^7.8.1` |
| `node`                      | `v13.12.0` |

**Comment by bradzacher** [comment]
You will have to setup your eslint config to use [`overrides`] (https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns) to disable type aware linting on HTML files.

Type aware linting requires TS files on disk. HTML files cannot fit that criteria.

**Comment by aress31** [comment]
So, in short, I cannot run my `eslint config` on `html files` is that correct? So how would you do if your `html files` include some `script tags` that you want to be processed by `eslint`?

**Comment by bradzacher** [comment]
You just can't use type-aware linting on HTML files.
All other parts of linting will work fine.

You can either use `overrides` to match HTML files and disable `parserOptions.project`, or you can use overrides to match TS files and enable `parserOptions.project`

You will also have to enable/disable type-aware lint rules in the same way.

**Comment by aress31** [comment]
Thank you so much @bradzacher for the explanations! Just being a bit lazy here but would you be kind enough to provide a minimal example? 😁

**Comment by aress31** [comment]
Answering my previous post, I fixed the issue as follows:
```JSON
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:lit/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "eslint-config-prettier"
  ],
  "overrides": [
    {
      "extends": [
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking"
      ],
      "files": ["*.ts"],
      "parser": "@typescript-eslint/parser",
      "parserOptions": {
        "project": ["./tsconfig.json"],
        "tsconfigRootDir": "."
      },
      "plugins": ["@typescript-eslint"]
    }
  ],
  "plugins": ["import", "html", "sort-keys-fix"],
  "root": true,
  "rules": {
    "import/no-unresolved": "off",
    "sort-keys-fix/sort-keys-fix": "error"
  }
}
```

Thanks again @bradzacher for your help! 👍

**Comment by aress31** [comment]
@bradzacher, dumb question - just to make sure that this configuration works as expected - when I override the configuration for the `*.ts` files, does the basic configuration still applies to `*.ts` files - extending the basic config just adding the extra parser options - or does the basic configuration get voided?

Or in other words, does the following configuration still applied to `ts` files?
```JSON
  "extends": [
    "eslint:recommended",
    "plugin:lit/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "eslint-config-prettier"
  ],
```

**Comment by bradzacher** [comment]
yes - `overrides` just applies additional config to the specified globs.
So existing config is still included in overridden files.
